src/include/kompute/logger/Logger.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <set>
+#include <spdlog/spdlog.h>
```

Comment:
Spdlog shoudl be an optional dependency as quite an opinionated logging library (only required dependency in package is fmt)

Reply:
Done. Now, when the user sets `KOMPUTE_OPT_LOG_LEVEL` to `Off`, we won't even fetch/build/link all the logger stuff.
https://github.com/KomputeProject/kompute/pull/287/commits/6f354fb40beebccf47558d904dd6f647f3e4d61f

Reply:
Ok thanks for having a look, however I don't think this woudl have the desired effect. The main component that should be optional, it's not that we should only include logging if spdlog is a direct dependency. Overall I think all the changes make sense, I'm not yet convinced of his Logger.hpp implementation, it would be good to dive into more detail on the value of having this approach. So far my thinking on this is:

* Logging should always work via KP_LOG_<DEBUG,INFO,etc...>
* SPDLOG should be an optional dependency
* Logging should still work if SPDLOG is disabled
* In the case of python builds, the setup would use python loggers
* In the case of standard builds, the logger would use just fmt (similar to how it was previously)
* Kompute shouldn't have an opinonated way in how logging is configured if someone uses spdlog

Based on these points it would be good to get more thoughts on the following:
* Why do we need to have a file in the core library that logs to files when spdlog provides this functionality (https://github.com/gabime/spdlog#rotating-files)? This means people would be able to configure the creation of files if they need to do it as part of their applications
* If we have an option for logging just with fmt and not with spdlog why would we implement this as toString instead of having a fmt configuration? Would it be assuming that people may prefer to just get the string implementation instead of using fmt? 

Based on the points above I am still leaning more towards:

* Not having a logger module with helpers (i.e. creating directory, logging to files, etc should all be on the application as opposed to on the core library)
* Having fmt as the basic way of logging, ensuring that spdlog is an optional dependency but logging is always available
* Using fmt formatters instead of toString - I do like the idea of allowing people converting to string in case they want to print outside fmt, however I don't see as much value if fmt continues to be the only requried dependency

Reply:
Done. This can now be controlled via `KOMPUTE_OPT_LOG_LEVEL` and `KOMPUTE_OPT_USE_SPDLOG`.
In case logging is enabled and we do not use Spdlog, we will log using fmt. I made sure that users also can override the logging macros when `KOMPUTE_OPT_USE_SPDLOG` is set to `OFF`.

Also I fixed logging for python and android again.

Reply:
Perfect, running a few initial tests